test/functional/mempool_datacarrier.py

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env python3
+# Copyright (c) 2020-2021 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test datacarrier functionality"""
+import os
+
+from test_framework.messages import CTxOut, CTransaction
+from test_framework.script import CScript, OP_RETURN
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.test_node import TestNode
+from test_framework.util import assert_raises_rpc_error
+from test_framework.wallet import MiniWallet
+
+
+class DataCarrierTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.num_nodes = 3
+        self.extra_args = [[], ["-datacarrier=false"], ["-datacarrier=true", "-datacarriersize=5"]]
+
+    def test_transaction(self, node: TestNode, tx: CTransaction, success: bool):
+        tx_hex = tx.serialize().hex()
+
+        if success:
+            self.wallet.sendrawtransaction(from_node=node, tx_hex=tx_hex)
+            assert tx.rehash() in node.getrawmempool(True), f'{tx_hex} not in mempool'
+        else:
+            assert_raises_rpc_error(-26, "scriptpubkey", self.wallet.sendrawtransaction, from_node=node, tx_hex=tx_hex)
+
+    def create_null_data_transaction(self, script: CScript) -> CTransaction:
+        tx = self.wallet.create_self_transfer(fee_rate=0)["tx"]
+        tx.vout.append(CTxOut(nValue=0, scriptPubKey=script))
+        tx.vout[0].nValue -= tx.get_vsize()  # simply pay 1sat/vbyte fee
+        return tx
+
+    def run_test(self):
+        # Default setting for -datacarriersize: 80 bytes of data.
+        MAX_OP_RETURN_DATA_SIZE = 80
+
+        self.wallet = MiniWallet(self.nodes[0])
+        self.wallet.rescan_utxos()
+
+        data = os.urandom(MAX_OP_RETURN_DATA_SIZE)
```

Comment:
Would suggest to use the deterministic `random_bytes` helper instead (to be found in `test_framework.util`). (Background: also used the `urandom` approach for generating random data recently in a PR and got a review with good counter-arguments: https://github.com/bitcoin/bitcoin/pull/25625#discussion_r924540431)

Reply:
Great suggestions! All have been addressed in 740db1b28a3922aef026e7c86c0bd78d95de1079.